packages/core/src/components/TextField/getMaskedValue.ts

```diff
@@ -1,29 +1,23 @@
-import type { ChangeEvent } from 'react';
-
-const getMaskedValue = (event: ChangeEvent<HTMLInputElement>, mask: string): string => {
-    const { value } = event.target,
+const applyMasking = (value: string, mask: string): string => {
+    const { length } = value,
+        lastChar = value[length - 1],
         alphaNumericRegex = /[a-zA-Z0-9]/,
-        specialCharsRegex = /[^a-zA-Z0-9]/,
-        { length } = value,
-        lastChar = value[length - 1];
+        specialCharsRegex = /[^a-zA-Z0-9]/;
 
     let newValue;
 
-    // if user deletes some char from middle
-    if (event.target.selectionStart && event.target.selectionStart < length) {
-        newValue = value.slice(0, event.target.selectionStart);
-    } else if (length > mask.length || /[a-zA-Z]/.test(lastChar)) {
+    if (length > mask.length || /[a-zA-Z]/.test(lastChar)) {
```

Comment:
Can we also move `/[a-zA-Z]/` this to a variable for better readability?

Reply:
Done. Nice finding. :)